src/marin/rl/rollout_worker.py

```diff
@@ -58,29 +62,55 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass
-class RolloutWorkerConfig:
-    """Configuration for RolloutWorker."""
+@dataclass(kw_only=True)
+class BaseRolloutWorkerConfig:
+    """Base configuration for RolloutWorker."""
 
-    inference_server_config: InferenceServerConfig
-    trainer: TrainerConfig
-    model: LmConfig
     curriculum_config: CurriculumConfig
     rollout_storage: RolloutStorageConfig
     weight_transfer: WeightTransferConfig
     tokenizer: PreTrainedTokenizer
     run_id: str
-
+    trainer: TrainerConfig
+    model: LmConfig
     seed: int = 0
     """Random seed to use for sampling."""
-
     max_rollouts: int | None = None
     """Maximum number of rollouts to generate before stopping. Defaults to running forever."""
 
+    log_freq: int = 10
+
+    inference_type: Literal["levanter", "vllm"] = "levanter"
+
     initial_checkpoint: str | None = None
     """Initial checkpoint for the reference model (auto-detects HF repo vs local path)."""
 
-    log_freq: int = 10
+
+@dataclass
+class LevanterRolloutWorkerConfig(BaseRolloutWorkerConfig):
+    """Configuration for RolloutWorker."""
+
+    inference_server_config: InferenceServerConfig
+
+
+@dataclass
+class VLLMRolloutWorkerConfig(BaseRolloutWorkerConfig):
+    """Configuration for RolloutWorker."""
+
+    vllm_model_name: str
```

Comment:
Nit, I'd recommend we move these configurations & the server management into the inference context, and then the rollout worker will just call the inference context when it has new weights for it etc.

(Obviously after we get everything working)